content/blog/hpe-morpheus-enterprise-case-study-fix-network-pool-links-via-a-custom-task-plugin.md

```diff
@@ -0,0 +1,158 @@
+
+![Provisioned VM to network](/img/morphblog_linknetwork_provisioned_vm.png)
+
+Next, we create a host entry manually within the pool:
+
+![New Pool IP](/img/morphblog_linknetworks_manual_host_record.png)
+
+Finally, we add the IP pool back onto the network:
+
+![Network pool re-added onto network](/img/morphblog_linknetwork_network_pool_readded.png)
+
+This time, we can see that the ***refId*** property is NOT populated on the ***ComputeServer***, despite the matching host name on the VM:
+
+![API host record with no link](/img/morphblog_linknetwork_api_host_record_nolink.png)
+
+## Compiling and loading the Custom Task Plugin
+
+Download or clone the plugin repository from <https://github.com/neilvrhpe/link-network-hosts>.
+Open the project directory and compile with the relevant ***gradlew*** (Linux) or ***gradlew.bat*** (Windows) script using the ***shadowJar*** argument:
+
+![Compile the plugin jar file](/img/morphblog_linknetworks_compile_jar.png)
+
+The compiled ***.jar*** file will be found in the ***build/libs*** subdirectory:
+
+![Compiled jar file in build/libs subdirectory](/img/morphblog_linknetworks_compiled_jar.png)
+
+Upload the ***.jar*** file to the ***Administration > Integrations > Plugins > Add*** dialog. The plugin should appear in the list as shown below:
+
+![Plugin uploaded and shown in list](/img/morphblog_linknetworks_plugin_uploaded.png)
+
+## Setup the Custom Task Workflow
+
+Uploading the plugin in the previous step introduced a new ***TaskType*** to the HPE Morpheus Enterprise appliance. This can be seen under the edit dialog of the uploaded plugin:
+
+![New task type introduced](/img/morphblog_linknetworks_new_tasktype.png)
+
+To use this new custom task type in the UI, we will provide an ***OptionSource*** input. We create the corresponding ***OptionList*** under ***Library > Options > Option Lists > Add***. The type is ***Plugin*** and the option list will be ***Link Network Hosts: getNetworkPools***, as provided by the plugin that we uploaded in the previous step. Provide 'ChooseNetworkPool' as the name, so that it matches the name in the next step:
+
+![Create option source](/img/morphblog_linknetworks_create_optionsource.png)
+
+Next, we create the ***Input*** that represents the ***OptionList*** entries to the end user dropdown in the UI, This is done using the ***Library > Options > Inputs > Add*** button. We will provide 'networkPool' as the ***Field Name*** for the custom task to reference in the next step. Choose ***Select List*** as the ***Type*** and use ***ChooseNetworkPool*** as the ***Option List*** field value:
+
+![Create the form input for the option source](/img/morphblog_linknetworks_form_input.png)
+
+To set up the task, we navigate to ***Library > Automation > Tasks > Add***. Provide a task ***Name*** and a ***Network Pool Id*** value of ***<%=customOptions.networkPool%>***. This reference will insert the value from the input we created in the previous step.
```

Comment:
Recommended rewrite:

"To set up the task, navigate to..."

"This reference will insert the value from the input created in the previous step."